AsyncDisplayKit/TextKit/ASTextKitFontSizeAdjuster.mm

```diff
@@ -179,23 +181,42 @@ - (CGFloat)scaleFactor
     
     if (_attributes.maximumNumberOfLines > 0) {
       // get the number of lines in our possibly scaled string
-      NSUInteger numberOfLines = [self lineCountForString:textStorage];
+      NSUInteger numberOfLines = [self lineCountForString:scaledString];
       if (numberOfLines > _attributes.maximumNumberOfLines) {
         
         for (NSUInteger index = scaleIndex; index < scaleFactors.count; index++) {
-          NSMutableAttributedString *entireAttributedString = [[NSMutableAttributedString alloc] initWithAttributedString:textStorage];
-          [[self class] adjustFontSizeForAttributeString:entireAttributedString withScaleFactor:[scaleFactors[index] floatValue]];
-          
+          scaledString = [[NSMutableAttributedString alloc] initWithAttributedString:textStorage];
```

Comment:
Can you create a new var here and add a comment that it needs to be reset each loop because it's being mutated below? Maybe the comment is unnecessary if you create a new var here…